crates/hir_ty/src/inference.rs

```diff
@@ -240,8 +240,9 @@ impl<'a> InferBody<'a> {
                 let param = param.data(self.hir_file.db(self.db));
                 self.infer_type(&param.ty)
             }
-            hir::Symbol::Local { name: _, expr } => {
-                let ty_scheme = self.current_scope().get(expr).cloned();
+            hir::Symbol::Local { name: _, binding } => {
+                let expr = binding.lookup(self.hir_file.db(self.db)).expr;
+                let ty_scheme = self.current_scope().get(&expr).cloned();
```

Comment:
**Optimization Issue**: The lookup operation for 'binding' and subsequent retrieval of 'expr' are performed separately, which can lead to redundant database queries and potential performance issues. <br> **Fix**: Combine the lookup of 'binding' and retrieval of 'expr' into a single operation to minimize database access and improve performance. <br> **Code Suggestion**: 
 ```
 - let expr = binding.lookup(self.hir_file.db(self.db)).expr;
 - let ty_scheme = self.current_scope().get(&expr).cloned();
 + let ty_scheme = self.current_scope().get(&binding.lookup(self.hir_file.db(self.db)).expr).cloned();
 ```